src/PendingObjects/TestCall.php

```diff
@@ -97,6 +97,8 @@ public function depends(string ...$tests): TestCall
             ->factoryProxies
             ->add(Backtrace::file(), Backtrace::line(), 'addDependencies', [$tests]);
 
+        $this->testCaseFactory->dependent = true;
```

Comment:
Can't you count dependencies internally?

Reply:
I tried doing that, but was not able to find where I can gain access to the `[$tests]` lists, the same we we have `$test->datasets`. Perhaps I missed something though. Where did you have mind?

Reply:
After some more looking, it seems that in order to count them locally, we would need to loop through the proxy message list, and check for `addDependencies` entries. If you don't think this will be an issue for performance (we make the check for every test and would have to loop over the whole message collection) we can try it, adding a method to the `HigherOrderMessageCollection` to get a count/bool?

Reply:
> adding a method to the HigherOrderMessageCollection to get a count/bool?

Looks good to me.

Reply:
Nice, I'll add and update later today.

Reply:
@nunomaduro Swapped out the prop set/check with the aforementioned `HigherOrderMessageCollection` method. Should be good for a review again whenever ya' get some minutes :)